### PR TITLE
Add a way to serialize a Swift object to JSValue object

### PR DESCRIPTION
**Pull request by ibratoev**
Serialize to a Dictionary and use that to pass it to JavaScriptCore JSValue. This seems to work fine and is simple to implement.

### CODE REVIEW

File: Sources/PipeableSDK/Script/SerializeToDictionary.swift [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        return dict
+    case .struct, .class:
+        return objectToDictionary(value)
+    default:
+        return value
+    }
+}
+
+// Serializes a struct or class to JSValue object.
+func serializeToJSValue(_ object: Any?, _ context: JSContext) throws -> JSValue {
```
Comment (pdobrev) [inline Sources/PipeableSDK/Script/SerializeToDictionary.swift:52]: This serializes only objects (structs, arrays, things with children).

Is the idea that we pass any value here and it's serialized to a JSValue (I think there was a function like this already somewhere) or is this just for objects. If it's just for objects, then I suggest renaming it.

**Review by pdobrev — APPROVED** [review]
Looks good, added one question

File: Sources/PipeableSDK/Script/SerializeToDictionary.swift (hunk 1 above)
Comment (ibratoev) [reply Sources/PipeableSDK/Script/SerializeToDictionary.swift:52]: Yeah, good point. I will add some tests specific to this function and document it better.

**Review by pdobrev — APPROVED** [review]

**Comment by pdobrev** [comment]
LGTM, maybe we should add a couple of tests for the trivial types (number, string, etc), but it's good.

**Comment by ibratoev** [comment]
> LGTM, maybe we should add a couple of tests for the trivial types (number, string, etc), but it's good.

You should not need to use serialization for simple types. But the serializer would work - yes. I will add a few tests. 
I considered making it not work for simple types but it will be more work and not really worth it.

**Pull request merged**